packages/worker/src/worker.ts

```diff
@@ -464,11 +464,7 @@ export class Worker {
     workflowBundle: WorkflowBundleWithSourceMapAndFilename,
     compiledOptions: CompiledWorkerOptions
   ): Promise<WorkflowCreator> {
-    const registeredActivityNames = new Set(
-      Object.entries(compiledOptions.activities ?? {})
-        .filter(([_, v]) => typeof v === 'function')
-        .map(([k]) => k)
-    );
+    const registeredActivityNames = new Set(compiledOptions.activities.keys());
```

Comment:
```suggestion
    const registeredActivityNames = [...new Set(compiledOptions.activities.keys())];
```

Reply:
Uhh? Why? The Creators just bellow expect Sets, not Arrays.

Reply:
Sorry, I should've read this more closely, when I skimmed the diff it looked like you were converting this set to an array in line 865.

Ignore me.